File: libs-node/fileClient.js
"use strict";
var path = require('path');
var s3 = require("./aws").S3;
var fs = require('fs');

function AWSFileClient() {

}

AWSFileClient.prototype.putFile = function(fileName, fileBody, folder, next) {
    var data = {
        Key: fileName,
        Body: fileBody,
        Bucket: folder
    };
    s3.putObject(data, function(err) {
        if (err) {
            next(err);
        } else {
            next();
        }
    });
};

function LocalFileClient() {

}

LocalFileClient.prototype.putFile = function(fileName, fileBody, folder, next) {
    fs.writeFile(path.join(folder, fileName), fileBody, function(err) {
        if (err) {
            next(err);
        } else {
            next();
        }
    });
};

function FileClient(provider) {
    if (provider === "AWS") {
        return new AWSFileClient();
    } else if (provider === "LOCAL") {
        return new LocalFileClient();
    }
}

module.exports = FileClient;

File: lwutils/file/file_client.py
from lwutils.file.clients.aws_file_client import AWSFileClient
from lwutils.file.clients.local_file_client import LocalFileClient


class FileClient(object):

    def factory(config):
        if config.get('FILE_PROVIDER') == "AWS":
            return AWSFileClient(config)
        elif config.get('FILE_PROVIDER') == "LOCAL":
            return LocalFileClient(config)
        else:
            print "Unknown File provider: " + config.get('FILE_PROVIDER', "not configured")
    factory = staticmethod(factory)
